model/src/main/kotlin/config/AnalyzerConfiguration.kt

```diff
@@ -36,10 +36,17 @@ data class AnalyzerConfiguration(
     val allowDynamicVersions: Boolean = false,
 
     /**
-     * Package manager specific configuration options. The key needs to match the name of the package manager class,
-     * e.g. "NuGet" for the NuGet package manager. See the documentation of the respective class for available options.
+     * A flag to indicate whether all package managers should be enabled by default. If true, all package managers that
+     * are not explicitly disabled are enabled. If false, all package managers that are not explicitly enabled are
+     * disabled.
      */
-    val options: Map<String, Options>? = null,
+    val enablePackageManagersByDefault: Boolean = true,
```

Comment:
Do we really need this new flag? I have a tendency to not introduce it to reduce complexity. On CLI, also all PMs are enabled by default without the possibility to change that.

Reply:
It's there for the same reason as the `-n` command line parameter. If you want to enable only one package manager, you would otherwise have to disable all others.

Reply:
I believe on the CLI being able to negate the logic made sense, as listing e.g. all but one package manager clutters the command line, and having to type the list of PMs over and over again is inconvenient. However, with the PM options in `ort.conf` I don't think that's a real issue anymore. You could simply search & replace if you'd want to "mass toggle" PMs.

That's why I'd still be in favor of removing that option completely. However, maybe there's an alternative: I know you already mentioned that you like your way of per-PM options. But many we could extract the `enabled` flag out of it, and still have my proposed lists of enabled / disabled PMs in `ort.conf`. This would be similar to e.g. the scanner storage configuration, where the storages themselves have their options, but whether they are actually used as readers / writers is configured separately.

Reply:
> I don't like the way it's implemented for scan storages either, because the configuration is scattered around.

It's maybe a question of encapsulation: Should a storage / PM itself know whether it's enabled, or is that the responsibility of an outer "manager"? I tend to say it's the latter.

> I would actually prefer if there were `read` and `write` flags in the storage configuration itself.

Even if getting a bit off topic: How would you then define the priority of storages?

Actually that question is semi-related to PMs: We'll soon have the need to configure the (relative) order in which PMs are run. I believe that requires a high-order concept anyway, similar to the mentioned enabled / disabled lists. Did you think about a syntax for that in `ort.conf` yet?

Reply:
> It's maybe a question of encapsulation: Should a storage / PM itself know whether it's enabled, or is that the responsibility of an outer "manager"? I tend to say it's the latter.

I think the focus for the config file should be usability, not software design patterns, and how the values of the configuration classes are mapped to other classes is an implementation detail. Also for a value like `licenseFilePatterns` in the top-level configuration the location does not tell anything about in which classes this value is used internally, only that it is a global configuration that is not specific to one ORT tool.

> > I would actually prefer if there were `read` and `write` flags in the storage configuration itself.
> 
> Even if getting a bit off topic: How would you then define the priority of storages?

I would say the order in which they are defined. That would work because the priority is only relevant for reading, not for writing.

> Actually that question is semi-related to PMs: We'll soon have the need to configure the (relative) order in which PMs are run. I believe that requires a high-order concept anyway, similar to the mentioned enabled / disabled lists. Did you think about a syntax for that in `ort.conf` yet?

What we discussed separately was the we will need a way to configure that a package manager must run after (or before) another, so it would be like:
```
packageManagers {
  Gradle.mustRunAfter = Maven, NPM
}
```
Or in a more complete example it could look like:
```
enablePackageManagersByDefault = false

packageManagers {
  NPM.enabled = true

  NuGet {
    enabled = true
    mustRunAfter = NPM
    options {
      directDependenciesOnly = true
    }
  }
}
```
Or alternatively a `mustRunBefore` option. From the functionality they would be equivalent, so I'm not sure if we should have both or only one of them, and then which.

Reply:
```
enablePackageManagersByDefault = false

packageManagers {
  NPM.enabled = true

  NuGet {
    enabled = true
    mustRunAfter = NPM
    options {
      directDependenciesOnly = true
    }
  }
}
``` 

Maybe one thing which hasn't been considered yet in above more complete example. If a package manager has configuration attributes without default values, then disabling a package manager (assuming `enablePackageManagersByDefault = true`) requires adding further configuration attributes. Probably this case isn't relevant as there are always default values, just wanted to bring it to consideration.

Apart from that we could add one more alternative implementation:
´´´
enabledPackageManagers = NPM, Yarn, Gradle
disabledPackageManager = GoMod
``` 
1. No default values for each package managers enabled state
2. Behavior: in case enabledPackageManagers + disabledPackageManager != allPackageManager: stop with an error sayinfg that the enabled state must be defined
3. This will enforce the user to make a conscious decision whether to enable any package manager after an upgrade of ORT which added a package manager

Reply:
> Maybe one thing which hasn't been considered yet in above more complete example. If a package manager has configuration attributes without default values, then disabling a package manager (assuming `enablePackageManagersByDefault = true`) requires adding further configuration attributes. Probably this case isn't relevant as there are always default values, just wanted to bring it to consideration.

I'm not sure if I understand that correctly, but maybe it's a misunderstanding. If the config contains a line like `NuGet.enabled = false` it does not mean that an instance of `NuGet` is created and therefore options without default values would have to be provided. The actual class that is instantiated is `PackageManagerConfiguration`.

> Apart from that we could add one more alternative implementation:
> 
> ```
> enabledPackageManagers = NPM, Yarn, Gradle
> disabledPackageManager = GoMod
> ```
> 
>     1. No default values for each package managers enabled state
> 
>     2. Behavior: in case enabledPackageManagers + disabledPackageManager != allPackageManager: stop with an error saying that the enabled state must be defined
> 
>     3. This will enforce the user to make a conscious decision whether to enable any package manager after an upgrade of ORT which added a package manager

That's a good point, but as my intention is to allow this configuration also in the `.ort.yml` in a later step, it would not be manageable.